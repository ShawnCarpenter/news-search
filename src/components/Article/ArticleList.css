.ArticleList {
  ul {
    max-width: 80vw;
  }
  li{
    list-style: none;
  }
}