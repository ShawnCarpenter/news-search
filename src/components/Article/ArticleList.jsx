import React from 'react';
import PropTypes from 'prop-types';
import Article from './Article';
import styles from './ArticleList.css';

const ArticleList = ({ articles }) => {
  const articleList = articles.map(article => (
    <li key={article.url}>
      <Article {...article} />
    </li>
  ));

  return <ul className={styles.ArticleList}>{articleList}</ul>;
  
};

ArticleList.propTypes = {
  articles:PropTypes.arrayOf(PropTypes.object)
};
export default ArticleList;
