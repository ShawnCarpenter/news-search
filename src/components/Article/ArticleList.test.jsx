import React from 'react';
import { render, cleanup } from '@testing-library/react';
import ArticleList from './ArticleList';

describe('ArticleList component', () => {
  afterEach(() => cleanup());
  it('renders ArticleList', () => {
    const { asFragment } = render(<ArticleList articles={[
      {
        title: 'Headline here',
        author: 'Clark Kent',
        description: 'Some stuff happened',
        url: 'https://www.dailyplanet.com',
        urlToImage: 'https://www.dailyplanet.com/img'
      }
    ]} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
