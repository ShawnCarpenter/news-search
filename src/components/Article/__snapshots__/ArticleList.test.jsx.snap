// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList component renders ArticleList 1`] = `
<DocumentFragment>
  <ul
    class="ArticleList"
  >
    <li>
      <div
        class="Article"
      >
        <h1>
          <a
            href="https://www.dailyplanet.com"
          >
            Headline here
          </a>
        </h1>
        <span>
          Author: Clark Kent
        </span>
        <img
          alt="Headline here"
          src="https://www.dailyplanet.com/img"
        />
        <div>
          Some stuff happened
        </div>
      </div>
    </li>
  </ul>
</DocumentFragment>
`;
